extensions/amp-apester-media/0.1/amp-apester-media.js

```diff
@@ -0,0 +1,232 @@
+/**
+ * Copyright 2016 The AMP HTML Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+import {CSS} from '../../../build/amp-apester-media-0.1.css';
+import {user} from '../../../src/log';
+import {loadPromise} from '../../../src/event-helper';
+import {Layout} from '../../../src/layout';
+import {isExperimentOn} from '../../../src/experiments';
+import {removeElement} from '../../../src/dom';
+import {vsyncFor} from '../../../src/vsync';
+
+/** @const */
+const TAG = 'amp-apester-media';
+
+/**
+ * AMP Apester-media
+ */
+class AmpApesterMedia extends AMP.BaseElement {
+  /** @override */
+  preconnectCallback(onLayout) {
+    this.preconnect.url(this.displayBaseUrl_, onLayout);
+    this.preconnect.url(this.rendererBaseUrl_, onLayout);
+  }
+
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == Layout.FIXED_HEIGHT;
+  }
+
+  /** @override */
+  viewportCallback(inViewport) {
+    if (inViewport && !this.seen_) {
+      if (this.iframe_.contentWindow) {
+        console.log('seen');
+        this.seen_ = true;
+        this.iframe_.contentWindow./*OK*/postMessage('interaction seen', '*');
+      }
+    }
+  }
+
+  /** @override */
+  buildCallback() {
+
+    // EXPERIMENT
+    user().assert(isExperimentOn(this.win, TAG), `Enable ${TAG} experiment`);
+
+    /**
+     * @const @private {?String}
+     */
+    this.rendererBaseUrl_ = 'https://renderer.qmerce.com';
+
+    /**
+     * @const @private {?String}
+     */
+    this.displayBaseUrl_ = 'https://display.apester.com';
+
+    /**
+     * @const @private {?String}
+     */
+    this.mediaAttribute_ = user().assert(
+      (this.element.getAttribute('data-apester-media-id') ||
+      (this.random = this.element.getAttribute('data-apester-channel-token'))),
+      'Either the data-apester-media-id or the data-apester-channel-token '
+      + 'attributes must be specified for <amp-apester-media> %s',
+      this.element);
+
+    /**
+     * @private {?Element}
```

Comment:
Actually, it looks to be constructed in the `layoutCallback`. Why is it also constructed here?